libsmtutil/Exceptions.h

```diff
@@ -0,0 +1,31 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libsolutil/Assertions.h>
+#include <libsolutil/Exceptions.h>
+
+namespace solidity::smtutil
+{
+
+struct SMTLogicError: virtual util::Exception {};
```

Comment:
It is possible new exceptions may need to be added to StandardCompiler and/or CompilerStack and other places, but likely not. In the case of Yul exceptions there was one case where it was needed.

Reply:
I don't think I get it, what would that be the case?
These occurrences use to be `solAssert`, they were just replaced.
The `solAssert` ones in `libsolidity/formal` stay the same.

Reply:
There is a new exception type thrown. If you have code which excepts the old exception type that will break.

Reply:
The mess I encountered was in #7884.

Reply:
The old exception type is still thrown everywhere